extensions/wikia/ImageReview/ImageReviewHelper.class.php

```diff
@@ -203,221 +235,283 @@ public function refetchImageListByTimestamp( $timestamp ) {
 	}
 
 	/**
-	 * get image list
+	 * Get image list
+	 *
+	 * @param $timestamp
+	 * @param int $state
+	 * @param int $order
+	 * 
 	 * @return array imageList
+	 *
+	 * @throws DBUnexpectedError
+	 * @throws Exception
+	 * @throws MWException
 	 */
 	public function getImageList( $timestamp, $state = ImageReviewStatuses::STATE_UNREVIEWED, $order = self::ORDER_LATEST ) {
-		wfProfileIn( __METHOD__ );
+		// Get the start date to use for all images collected here
+		$reviewStart = wfTimestamp( TS_DB, $timestamp );
 
-		$oDB = $this->getDatawareDB( DB_MASTER );
-		$oDatabaseHelper = $this->getDatabaseHelper();
+		$deleteFromQueueList = [];
+		$imageList = [];
+		$iconsWhere = [];
 
-		$oResults = $oDatabaseHelper->selectImagesForList( $this->getOrder( $order ), self::LIMIT_IMAGES_FROM_DB, $state );
+		// Set a hard and soft limit for stopping the loop.  The soft limit changes
+		// only when an image matches, the hard limit changes for every iteration.
+		$hardNum = self::LIMIT_IMAGES_FROM_DB;
+		$softNum = self::LIMIT_IMAGES;
 
-		$rows = array();
-		$updateWhere = array();
-		$iconsWhere = array();
-		while ( $row = $oDB->fetchObject($oResults) ) {
-			$rows[] = $row;
-			$updateWhere[] = "(wiki_id = {$row->wiki_id} and page_id = {$row->page_id})";
-		}
-		$oDB->freeResult( $oResults );
-
-		# update records
-		if ( count($updateWhere) > 0) {
-			$review_start = wfTimestamp( TS_DB, $timestamp );
-
-			switch ( $state ) {
-				case ImageReviewStatuses::STATE_QUESTIONABLE:
-					$target_state = ImageReviewStatuses::STATE_QUESTIONABLE_IN_REVIEW;
-					break;
-				case ImageReviewStatuses::STATE_REJECTED:
-					$target_state = ImageReviewStatuses::STATE_REJECTED_IN_REVIEW;
-					break;
-				default:
-					$target_state = ImageReviewStatuses::STATE_IN_REVIEW;
+		$imageIter = $this->getLatestImageIter( $order, $state );
+
+		while ( $softNum && $hardNum && $row = $imageIter() ) {
+			$hardNum--;
+
+			// Grab this image now to eliminate any race conditions with other reviewers
+			if ( !$this->assignImageToUser( $reviewStart, $state, $row ) ) {
+				// If we failed to assign this image to ourselves, try the next
+				continue;
 			}
 
-			$values = array (
-				'reviewer_id' => $this->user_id,
-				'review_start' => $review_start,
-				'state' => $target_state
-			);
+			$imageInfo = $this->getImageData( $row );
 
-			if ( $state == ImageReviewStatuses::STATE_QUESTIONABLE || $state == ImageReviewStatuses::STATE_REJECTED ) {
-				$values[] = "review_end = '0000-00-00 00:00:00'";
+			// If we failed to load this image, remove it from the review queue
+			if ( !empty( $imageInfo['error'] ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "error" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)